help: Document "Start a new topic" mobile feature.

drrosa opened the pull request:
- Adds mobile instructions block to the [Starting a new topic](https://zulip.com/help/starting-a-new-topic) page.
- Deletes [Start a new topic](https://zulip.com/help/starting-a-new-topic) page in favor of "Starting a new topic".

**Screenshots and screen captures:**

![image](https://github.com/zulip/zulip/assets/2343554/19641715-1346-42ff-a43d-e761df9a2d84)

![image](https://github.com/zulip/zulip/assets/2343554/a8dcd405-be04-497c-843b-a5a08104616d)

![image](https://github.com/zulip/zulip/assets/2343554/e90151f7-9d78-43f6-a7fa-b3e6ac0aea91)

<details>
<summary>Self-review checklist</summary>

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability

Communicate decisions, questions, and potential concerns.

- [x] Automated tests.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
</details>

drrosa commented:
@laurynmm This PR is ready for review.
@zulipbot add "help center review"

laurynmm commented:
Hmm... this doesn't seem accurate for me, at least in the Andriod app. I don't see the compose box at the bottom of the app until I'm in a specific stream view. The streams view only has a list of streams.

drrosa commented:
@laurynmm Thanks! I updated the PR and screenshots.

> I don't see the compose box at the bottom of the app until I'm in a specific stream view. The streams view only has a list of streams.

Yeah, that's correct. I think I just got distracted and didn't include that second step in my commit. 😅

Also, I realized we have [Starting a new topic](https://zulip.com/help/starting-a-new-topic) too, so I updated the rest of the "Start a new topic" article for consistency.

laurynmm commented:
@drrosa - I figured it was something like that. These updates look good to me. Marking for integration review.

alya commented:
I think we should just delete the "Start a new topic" page. I like the "Starting a new topic" page better, and I don't see a reason to have both.

alya commented:
It's a bit non-standard, but maybe we can just keep the tabs we currently have on that page, and add a third tab for mobile instructions.

drrosa commented:
@zulipbot add "help center review"

alya commented:
Looks good to me!
